### DISCUSSION TITLE
Making local files temporarily accessible online?

### DISCUSSION THREAD

**Discussion by allefeld** [abraunegg/onedrive#2674] [General]
I'm not sure whether this is feasible at all, but I thought I'd ask:

I keep my files for work in local directories on a Linux machine. Most of the time, Linux software is sufficient to work with the files. However, every now and then I need something proprietary like Word, which I can use either in the browser version or within an online Windows virtual machine, both services provided by my employer, both with access to OneDrive / SharePoint files. In such cases, it would be convenient to temporarily make a local directory accessible online; if you will, "mount" the local directory within OneDrive / SharePoint. I would then use e.g. online / virtual Word to edit these files, after which I stop the connection, and the edited file remains in its original local directory.

As far as I can tell, the available tools (`rclone` for mounting, your `onedrive` for sync) make it either necessary to keep the files in online directories (folders) and temporarily mount them locally, or continuously synchronize directories. In one case, my files don't "live" locally anymore, and in the other I get the usual problems which inevitably come with synchronization. Or, I could have a special mounted or synchronized directory, but then I would have to copy files to and from the directory where I keep the files. (What I currently do is upload the files through the OneDrive web interface and download afterwards.)

Do you have an idea how to achieve (an approximation of) what I'm looking for?

**Comment by abraunegg** [comment]
@allefeld 
> and in the other I get the usual problems which inevitably come with synchronization.

Great effort has been put in to ensure items are synchronised correctly. Please review the architecture documentation to assist your understanding of how the client operates: https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/client-architecture.md

> Do you have an idea how to achieve (an approximation of) what I'm looking for?

Use the client as per 'normal' ... store your 'data where ever' locally, then symbolically link that local folder into your local OneDrive folder and that data will sync to OneDrive online and remain in sync until you remove the local symbolic link.

This allows you to:
* Edit the file locally to your hearts content
* Edit the file online, with changes synced back to your local system based on the 'monitor_interval' value (the time period the client checks for online content change)
  * If this is too in-frequent, configure a 'webhook' to listen for online changes realtime - please read https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-1/docs/usage.md#how-to-receive-real-time-changes-from-microsoft-onedrive-service-instead-of-waiting-for-the-next-sync-period

**Reply by allefeld** [reply]
> Great effort has been put in to ensure items are synchronised correctly.

I didn't mean to question the quality of your implementation. It's just that in my experience, even with the best software problems occur, so I'm wary to synchronize anything which I don't have to.

> Use the client as per 'normal' ... store your 'data where ever' locally, then symbolically link that local folder into your local OneDrive folder and that data will sync to OneDrive online and remain in sync until you remove the local symbolic link.

Great, that's an easy solution. Thanks!
